### PR TITLE
docs(readme): API documentation overhaul

### LINKED ISSUE #441: Rework the API Documentation so that it's more clear.
The API documentation hasn't changed much since I threw it together with the original module. Clearer documentation has always been a source of irritation for users.

### PR DESCRIPTION
**Pull request by rharshit82**
## Description

Refactored the Readme file into a more readable and standard readme

## Related Issue

Closes #441.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] If my change introduces a breaking change, I have added a `!` after the type/scope in the title (see the Conventional Commits standard).

### CODE REVIEW

**Comment by sheerlox** [comment]
great PR, the rewording is very good!

I've created a commit on my fork to put all the suggestions I had (seemed easier than writing down everything in a review).

you can view the diff here: https://github.com/rharshit82/node-cron/compare/fix/readme-overhaul...sheerlox:node-cron:docs/readme/api-documentation-overhaul

if you'd like to integrate my commit in your branch, you can pick the patch file and apply it in your branch:
```bash
curl https://github.com/sheerlox/node-cron/commit/1d22cf136b5596d4ae2e43fed35d427b3c78579b.patch | git am
```

you can also just apply the changes (without creating the commit) by piping the patch into `git apply` instead of `git am`.

**Review by sheerlox — CHANGES_REQUESTED** [review]
great PR, the rewording is very good!

I've created a commit on my fork to put all the suggestions I had (seemed easier than writing down everything in a review). all changes are obviously open to discussion!

you can view the diff here: https://github.com/rharshit82/node-cron/compare/fix/readme-overhaul...sheerlox:node-cron:docs/readme/api-documentation-overhaul

if you'd like to integrate my commit in your branch, you can pick the patch file and apply it in your branch:
```bash
curl https://github.com/sheerlox/node-cron/commit/1d22cf136b5596d4ae2e43fed35d427b3c78579b.patch | git am
```

you can also just apply the changes (without creating the commit) by piping the patch into `git apply` instead of `git am`.

**Comment by rharshit82** [comment]
Thanks @sheerlox  I have added your commit. I was actually going to fix table of contents and refactor the readme more but your commit as done just that. Marking this PR for review !

**Review by sheerlox — APPROVED** [review]

**Comment by sheerlox** [comment]
lgtm, I'll let @intcreator take a look as well so he can make changes as he sees fit before merging.

Note: I've added two badges that are not yet functional: the CodeQL one will start working when #715 is merged, and the Discord badge needs [a tweak in the Discord server settings (enable widget)](https://discord.com/blog/add-the-discord-widget-to-your-site/) to display the number of online members correctly.

**Comment by rharshit82** [comment]
Sure added it. Didnt add it before since I thought its only for a code change that required a doc change.

**Review by intcreator — CHANGES_REQUESTED** [review]
I have a few questions/requested changes, but overall I like it! I especially like...

- addition of Discord badge (I enabled the widget so it shows a preview of online people too)
- table of contents

(and I was just joking about the checklist item haha)

File: README.md
Comment (intcreator) [reply README.md:14]: nvm saw sheerlox's comment

File: README.md
Comment (rharshit82) [reply README.md:9]: This was changed in @sheerlox 's commit so he can respond here.

File: README.md
Comment (rharshit82) [reply README.md:63]: done

File: README.md
Comment (sheerlox) [reply README.md:9]: - the "Build Status" & "Build Checks" are merged in the single "Build Status" badge
- minified / minzipped sizes seemed irrelevant because the library is for Node.js, where most of the time package sizes aren't a concern, and when it is (for example in lambda cloud functions) the relevant metric is the unpacked size (which is available on the npm page)
 
regarding the Snyk badge, we (maintainers) have no access to its alerts, and the badge is static and doesn't provide more information than "monitored" / "unknown", as you can see below for a 10-year-old branch of `express`:
[Snyk](https://snyk.io/test/github/expressjs/express/benchmark/badge.svg)
image
and here's the head of the result of running [`osv-scanner`](https://osv.dev/) in that branch:
image

if we want to re-introduce that badge, here's an equivalent (static) badge in green and using the Shields project so it fits better with the other badges, wdyt?
[Vulnerabilities](https://img.shields.io/badge/vulnerabilities-monitored-green)

we could also replace it with a Renovate badge since it does the same job of finding vulnerabilities in dependencies:
[Renovate](https://img.shields.io/badge/renovate-enabled-green)

File: README.md
Comment (sheerlox) [reply README.md:34]: VSCode markdown rendered doesn't understand them, but they are working for me [on GitHub](https://github.com/kelektiv/node-cron/blob/3647be437505dabd0f73dd226eb1cba59573adc9/README.md)

File: README.md
Comment (intcreator) [reply README.md:63]: I'm not seeing the update at the moment, did you commit and push?

File: README.md
Comment (rharshit82) [reply README.md]: done

File: README.md
Comment (rharshit82) [reply README.md]: pushed now

**Comment by rharshit82** [comment]
Have made the requested changes. Feel free to make commits in this PR if needed @sheerlox

**Review by intcreator — DISMISSED** [review]
I just added the Renovate badge. I think that's everything, looks good to me

**Review by sheerlox — APPROVED** [review]

**Comment by sheerlox** [comment]
since we're planning on merging #751 today, I skipped the CI so we release them together under `3.1.5`.

**Pull request merged**
